### ISSUE TITLE
Call for Contributors: Train and Evaluate AstroVQA (maintainer hardware-limited)

### ISSUE THREAD

**Issue by Franz-kingstein** [Franz-kingstein/AstroVQA-#1]
Hi all,

I’m looking for contributors to help train, evaluate, and harden this Visual Question Answering (VQA) project for Earth Observation. Due to limited hardware, the model has not been fine-tuned yet. The codebase is ready; community help is needed to run experiments and improve it.

What’s needed
- Training
  - Run full/partial fine-tuning of Nano-LLaVA (LoRA + 4-bit) on the Indian EO VQA dataset.
  - Share configs, logs, and resulting checkpoints.
- Evaluation
  - Create evaluation scripts and report metrics (accuracy/bleu/rouge; open to suggestions).
  - Add a small validation set with reproducible results.
- Optimization
  - Memory/perf tweaks (gradient checkpointing, batch sizes, AMP, PEFT targets).
  - Optional: try SmolVLM as a lighter baseline.
- Datasets
  - Host datasets (GitHub Releases or Hugging Face Datasets).
  - Provide a robust download script and checksum verification.
- Docs and UX
  - Improve README (quickstart, Colab guide, dataset instructions).
  - Add an example inference notebook.
  - Optional: Dockerfile/devcontainer and CI sanity checks.

Getting started
- Repo: https://github.com/Franz-kingstein/AstroVQA-
- Setup:
  - Python 3.8+, CUDA GPU recommended
  - pip install -r requirements.txt
  - export HF_TOKEN=your_token
- Data:
  - Large datasets are excluded from git.
  - Either regenerate via create_eo_vqa.py and generate_final_vqa.py
  - Or help host on HF/Release and wire scripts/download_data.sh

Suggested tasks
- [ ] Run a first training on ≥24 GB GPU and report memory/throughput
- [ ] Add evaluation script + metrics and a small validation report
- [ ] Publish a trained LoRA checkpoint and usage instructions
- [ ] Host dataset on Hugging Face Datasets (or GitHub Release) and update scripts
- [ ] Add an inference demo notebook
- [ ] Improve README: quickstart, dataset, troubleshooting
- [ ] Optional: Dockerfile or devcontainer
- [ ] Optional: basic CI (lint, import test)

Please include in PRs/issues
- GPU type, VRAM, driver/CUDA versions
- Exact command line and config
- Logs, metrics, and any OOM/errors

Labels: help wanted, good first issue, documentation, dataset, research

Thanks to everyone willing to contribute!
